crates/node/engine/src/task_queue/tasks/build/error.rs

```
@@ -84,3 +85,19 @@ impl EngineTaskError for BuildTaskError {
         }
     }
 }
+
+/// The inter-actor error returned to the initial requestor of a Build.
+#[derive(Debug, Error)]
+pub enum BuildError {
+    /// A critical engine error occurred.
+    #[error("Error communicating with engine.")]
+    CommunicationError,
+
+    /// A critical engine error occurred.
+    #[error("A critical engine error occurred.")]
+    EngineError,
```

nit: it would be nice here if we could propagate the engine errors (`CommunicationError` and `EngineError`) here to log it when we're propagating the error.